refactor(filtered-search): split single and multi select

spike-rabbit opened the pull request:
This PR serves two goals:
1. Decouple logic of different criteria types. It helps to prevent unwanted sideeffect of changes and makes the code more readable.
2. Preparing the move of multi select logic from the typeahead to the filtered-search. As discussed internally, the typeahead should have never gotten multi-select support which anyway only works in the filtered-search.

> Describe in detail what your merge request does and why. Add relevant
> screenshots and reference related issues via `Closes #XY` or `Related to #XY`.

---

- [x] I confirm that this MR follows the [contribution guidelines](https://github.com/siemens/element/blob/main/CONTRIBUTING.md).

spliffone commented:
A general observation, clicking on the value area (e.g. si-filtered-search-typeahead) when the value not defined we do not switch to an input field. Is this behaviour by intention?

spliffone commented:
When I set `onlySelectValue` = `true` I would assume that the typing is disabled only for input fields which aren't multi select criteria. Since the multi-select input is used to filter criteria options. But this is right now not possible, what is your opinion?

spliffone commented on projects/element-ng/filtered-search/values/typeahead/si-filtered-search-typeahead.component.ts:
```diff
@@ -173,122 +87,39 @@ export class SiFilteredSearchTypeaheadComponent
-      optionsStream = of(toOptionCriteria(this.definition().options));
+  protected valueFilterKeys(event: KeyboardEvent): void {
+    if (
+      this.definition().validationType === 'integer' &&
```
A comment could be great, why du we filter certain key und certain conditions

spliffone commented:
I also noticed we dropped the behaviour that consumers can simply enter the value of an option e.g. CH results in Switzerland when pressing Enter. I guess we no longer want this behaviour, right?

spike-rabbit commented on projects/element-ng/filtered-search/values/si-filtered-search-option-value.base.ts:
```diff
@@ -0,0 +1,109 @@
+  protected readonly translateService = inject(SiTranslateService);
+
+  readonly inputType = computed(() =>
+    this.definition().validationType === 'integer' || this.definition().validationType === 'float'
```
I would not do this change. It has in theory worse performance and I don't think it adds that much of a readability benefit.

spike-rabbit commented on projects/element-ng/filtered-search/values/si-filtered-search-option-value.base.ts:
```diff
@@ -0,0 +1,109 @@
+    input<(criterionName: string, typed: string | string[]) => Observable<OptionType[]>>();
+  readonly searchDebounceTime = input.required<number>();
+  readonly itemCountText = input.required<string>();
+  readonly items = input.required<string>();
```
`items` was fully remove in the meantime, so no longer an issue

spike-rabbit commented on projects/element-ng/filtered-search/si-filtered-search-value.component.ts:
```diff
@@ -78,7 +80,11 @@ export class SiFilteredSearchValueComponent implements OnInit {
         return 'date';
       default:
-        return 'typeahead';
+        if (this.definition().multiSelect) {
```
done

spike-rabbit commented:
> I also noticed we dropped the behaviour that consumers can simply enter the value of an option e.g. CH results in Switzerland when pressing Enter. I guess we no longer want this behaviour, right?

Dropped on purpose ah while ago.

All other comments are valid. I will do in a follow-up.

spike-rabbit commented:
> When the user choose a multi-select criteria and select for example two items and either focus or click on the free text search the selected items aren't displayed. For example select two items in the si-filtered-search-playground of the criteria 'Score'

solved by rebasing. Please have a look again
